src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1232,6 +1232,86 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
+                "BCP212",
+                $"The Template Spec module alias name \"{aliasName}\" does not exist in the Bicep configuration \"{configurationResourceName}\".");
+
+            public ErrorDiagnostic OciArtifactModuleAliasNameDoesNotExistInConfiguration(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP213",
+                $"The OCI artifact module alias name \"{aliasName}\" does not exist in the Bicep configuration \"{configurationResourceName}\".");
+
+            public ErrorDiagnostic InvalidTemplateSpecAliasSubscriptionNullOrUndefined(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP214",
+                $"The Template Spec module alias \"{aliasName}\" in the Bicep configuration \"{configurationResourceName}\" is in valid. The \"subscription\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidTemplateSpecAliasResourceGroupNullOrUndefined(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP215",
+                $"The Template Spec module alias \"{aliasName}\" in the Bicep configuration \"{configurationResourceName}\" is in valid. The \"resourceGroup\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidOciArtifactModuleAliasRegistryNullOrUndefined(string aliasName, string configurationResourceName) => new(
+                TextSpan,
+                "BCP216",
+                $"The OCI artifact alias \"{aliasName}\" in the Bicep configuration \"{configurationResourceName}\" is invalid. The \"registry\" property cannot be null or undefined.");
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceInvalidSubscirptionId(string subscriptionId, string referenceValue) => new(
+                TextSpan,
+                "BCP217",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid. The subscription ID \"{subscriptionId}\" in is not a GUID."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceResourceGroupNameTooLong(string resourceGroupName, string referenceValue, int maximumLength) => new(
+                TextSpan,
+                "BCP218",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid. The resource group name \"{resourceGroupName}\" exceeds the maximum length of {maximumLength} characters."); 
+
+            public ErrorDiagnostic InvalidTemplateSpecReferenceInvalidResourceGroupName(string resourceGroupName, string referenceValue) => new(
+                TextSpan,
+                "BCP219",
+                $"The specified Template Spec reference \"{referenceValue}\" is invalid .The resource group name \"{resourceGroupName}\" is invalid. Valid characters are alphanumeric, \".\", \"_\", \"-\", \"(\", or \")\", but the resource group name cannot end with \".\"."); 
```

Comment:
>alphanumeric [](http://example.com/codeflow?start=170&length=12)

Don't RGs support unicode chars?

Reply:
Updated the validation logic and error message to handle unicode chars.